var/spack/repos/builtin/packages/pgi/package.py

```diff
@@ -64,6 +64,10 @@ class Pgi(Package):
     license_vars = ['PGROUPD_LICENSE_FILE', 'LM_LICENSE_FILE']
     license_url = 'http://www.pgroup.com/doc/pgiinstall.pdf'
 
+    def url_for_version(self, version):
+        return "file://{0}/pgilinux-20{1}-{2}-x86_64.tar.gz".format(
+            os.getcwd(), version.up_to(1), version.joined)
+
```

Comment:
We need a more systematic approach than this.  Can the user place it in a manual mirror, rather than CWD?  (I suppose that such behavior is not prohibited in this case).